cmake/setup_cached_variables.cmake

```diff
@@ -129,6 +129,17 @@ IF( NOT "${CMAKE_BUILD_TYPE}" STREQUAL "Release" AND
 ENDIF()
 
 #
+# Force exactly one build type for multi-cofiguration generators.
```

Comment:
Some generators (xcode, msvc) allow you to choose between different build configurations inside the IDE, in contrast to a single configuration like we have in makefiles, eclipse, etc..
Only the configuration selected is then built/executed/etc. But it turns out we never supported this correctly with the deal.II library. It does kind of work for projects/deal.II examples.